From 503c6777cc0a05e4ba174ff674fdf9ce3ed87341 Mon Sep 17 00:00:00 2001
From: joshuazivkovic <joshua.zivkovic@codethink.co.uk>
Date: Fri, 13 Jan 2023 09:17:27 +0000
Subject: [PATCH] systemd-analyze: Add --json=, --table and -no-legend tests
 for plot

(cherry picked from commit a23be57de3a5c5afb9ca878775ae838c3341f90c)

Resolves: RHEL-5070
---
 test/units/testsuite-65.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/test/units/testsuite-65.sh b/test/units/testsuite-65.sh
index ebe1f57b52..4093c5a2a7 100755
--- a/test/units/testsuite-65.sh
+++ b/test/units/testsuite-65.sh
@@ -18,7 +18,16 @@ systemd-analyze || :
 systemd-analyze time || :
 systemd-analyze blame || :
 systemd-analyze critical-chain || :
+# plot
 systemd-analyze plot >/dev/null || :
+systemd-analyze plot --json=pretty >/dev/null || :
+systemd-analyze plot --json=short >/dev/null || :
+systemd-analyze plot --json=off >/dev/null || :
+systemd-analyze plot --json=pretty --no-legend >/dev/null || :
+systemd-analyze plot --json=short --no-legend >/dev/null || :
+systemd-analyze plot --json=off --no-legend >/dev/null || :
+systemd-analyze plot --table >/dev/null || :
+systemd-analyze plot --table --no-legend >/dev/null || :
 # legacy/deprecated options (moved to systemctl, but still usable from analyze)
 systemd-analyze log-level
 systemd-analyze log-level "$(systemctl log-level)"
